README.md

```diff
@@ -12,28 +12,59 @@ The connector acts as a bridge between your Python application, which is typical
 * That user's password
 
 ## Authentication Environment
-In order to avoid passing authentication credentials in via command line arguments, Visier recommends that basic authentication credentials such as username and password are provided via environment variables. 
+As of version 0.9.8, the Visier Python Connector supports two means of authentication:
+1. OAuth2.0: The connector supports the so called three-legged authentication flow. This means that authentication (and consent) have to be provided through the authorization server's authentication mechanism. In accordance with the OAuth2.0 protocol, no user credentials are provided directly to Visier. This is the preferred authentication method.
+1. Basic Authentication: This is a traditional authentication mechanism where Visier username and password are provided directly to Visier for authentication.
 
-Though the Visier Python Connector doesn't directly interact with the environment variables, the following list and example below illustrate the various authentication parameters
+In order to avoid passing authentication credentials in via command line arguments, Visier recommends that at least basic authentication credentials such as username and password are provided via environment variables. However, using a new function, `auth_from_env()`, the appropriate authentication configuration object will be created from `VISIER_`-prefixed environment variables. Details are provided below.
+
+### OAuth2.0
+Though the Visier Python Connector doesn't directly interact with the environment variables, the following list and example below illustrate the OAuth2.0 authentication parameters. These are also the environment variables `auth_from_env()` will use.
+* `VISIER_HOST`: The fully qualified domain name and protocol to access your Visier tenant
+* `VISIER_APIKEY`: The API key granted by Visier
+* `VISIER_CLIENT_ID`: The identifier of the pre-registered application
+* `VISIER_REDIRECT_URI`: The URI the `authorize` call will ultimately redirect to upon a successful authorization code generation. By default, this will be `http://localhost:5000/oauth2/callback` however note that it must match the `redirect_uri` in the client application definition exactly. If the client application setting is different, it is essential that that exact value is provided in this environment variable.
+* `VISIER_TARGET_TENANT_ID`: The technical name of the tenant for the customer. This is only applicable in partner configurations
+
+On Linux-like systems, with an X-display available, create a file named `.env` and populate it like the following example, substituting with actual values as appropriate:
+```sh
+export VISIER_HOST=https://customer-specific.api.visier.io
+export VISIER_CLIENT_ID=client-id
+export VISIER_APIKEY=the-api-key-issued-by-visier
+export VISIER_REDIRECT_URI=
+# export VISIER_REDIRECT_URI=http://localhost:5000/oauth2/callback
```

Comment:
Should this be left in?

Reply:
Yes, as it shows what the default value is and where to change it if the default isn't acceptable.